packages/table/src/table.tsx

```diff
@@ -432,13 +449,23 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
 
     private columnHeaderCellRenderer = (columnIndex: number) => {
         const props = this.getColumnProps(columnIndex);
+        let loading: boolean;
+        switch (this.hasLoadingOption(props.loadingOptions, ColumnLoadingOption.HEADER)) {
```

Comment:
how is that not the same as this? if `this.hasLoadingOption()` returns `undefined` then `let loading` never gets set so it is also `undefined`. in my proposal you at least intentionally assign `undefined` to it.